collections/nemo_nlp/nemo_nlp/data/datasets/utils.py

```diff
@@ -778,21 +787,210 @@ def process_dialogflow(
 
-    write_df_files(train_queries, f'{outfold}/train.tsv')
-    write_df_files(train_slots, f'{outfold}/train_slots.tsv')
+    write_files(train_queries, f'{outfold}/train.tsv')
+    write_files(train_slots, f'{outfold}/train_slots.tsv')
 
-    write_df_files(test_queries, f'{outfold}/test.tsv')
-    write_df_files(test_slots, f'{outfold}/test_slots.tsv')
+    write_files(test_queries, f'{outfold}/test.tsv')
+    write_files(test_slots, f'{outfold}/test_slots.tsv')
 
-    write_df_files(slot_labels, f'{outfold}/dict.slots.csv')
-    write_df_files(intent_names, f'{outfold}/dict.intents.csv')
+    write_files(slot_labels, f'{outfold}/dict.slots.csv')
+    write_files(intent_names, f'{outfold}/dict.intents.csv')
 
     return outfold
 
 
-def write_df_files(data, outfile):
+def readCSV(file_path):
+    rows = []
+    with open(file_path, 'r') as csvfile:
+        readCSV = csv.reader(csvfile, delimiter=',')
+        for row in readCSV:
+            rows.append(row)
+    return rows
+
+
+def get_intents_mturk(utterances, outfold):
+
+    intent_names = {}
+    intent_count = 0
+
+    agreedallDict = {}
+
+    print('Printing all intent_labels')
+    intent_dict = f'{outfold}/dict.intents.csv'
+    if os.path.exists(intent_dict):
+        with open(intent_dict, 'r') as f:
+            for intent_name in f.readlines():
+                intent_names[intent_name.strip()] = str(intent_count)
+                intent_count += 1
+    print(intent_names)
+
+    for i, utterance in enumerate(utterances[1:]):
+
+        if utterance[1] not in agreedallDict:
+            agreedallDict[utterance[0]] = utterance[1]
+
+        if utterance[1] not in intent_names:
+            intent_names[utterance[1]] = str(intent_count)
+            intent_count += 1
+
+    print('x eval mechanism:')
+    print(len(agreedallDict))
+
+    return agreedallDict, intent_names
+
+
+def get_slot_labels(slot_annotations, task_name):
+
+    slot_labels = json.loads(slot_annotations[0])
+
+    all_labels = {}
+    count = 0
+    # Generating labels with the IOB format.
+    for label in slot_labels[task_name]['annotations']['labels']:
+        b_slot = ''.join(["B-", label['label']])
+        i_slot = ''.join(["I-", label['label']])
+        all_labels[b_slot] = str(count)
+        count += 1
+        all_labels[i_slot] = str(count)
+        count += 1
+    all_labels['O'] = str(count)
+
+    return all_labels
+
+
+def process_intent_slot_mturk(slot_annotations, agreedallDict, intent_names,
+                              task_name):
+
+    slot_tags = []
+    inorder_utterances = []
+    all_labels = get_slot_labels(slot_annotations, task_name)
+    print(f'agreedallDict - {len(agreedallDict)}')
+    print(f'Slot annotations - {len(slot_annotations)}')
+
+    for annotation in slot_annotations[0:]:
+        an = json.loads(annotation)
+        utterance = an['source']
+        if utterance.startswith('"') and utterance.endswith('"'):
```

Comment:
make sure that len(utterance) > 2

Reply:
Why '> 2' ?

Reply:
because if <= 2 then you don't need to check if it starts with " and end with "